.storybook/blocks/ComponentDetails.jsx

```diff
@@ -119,6 +157,188 @@ const VersionDetails = ({ tag, data = {}, isDeprecated = false, skipDate = false
+			</svg>
+		</ResourceIconWrapper>
+	)
+};
+
+/**
+ * Determines which SVG to use based on the resource link
+ */
+const iconSvgs = (linkType) => {
+	let svgAsset;
+
+	switch(linkType) {
+		case "package": 
+			svgAsset = npmSvg()
+			break;
+		case "repository": 
+			svgAsset = githubSvg()
+			break;
+		case "guidelines": 
+			svgAsset = adobeSvg()
+			break;
+		default: 
+			break;
+	}
+
+	return svgAsset;
+};
+
+/**
+ * Converts the linkType to the subtitle text
+ */
+const convertLinkTypeText = (linkType) => {
+	switch(linkType) {
+		case "package": 
+			linkType = "npm"
+			break;
+		case "repository": 
+			linkType = "GitHub"
+			break;
+		case "guidelines": 
+			linkType = "Spectrum website"
+			break;
+		default: 
+			break;
+	}
+	return linkType;
+};
+
+export const ResourceLinkContent = ({data, linkType=["package", "repository", "guidelines"]}) => {
+	const packageJson = data;
+
+	// componentGuidelinesName
+	let packageName = "";
+	let packageLink = "";
+
+	// componentBetaName: for components with guidelines that are still on spectrum-contributions/beta site
+	let packageAltName = "";
+	let packageAltLink = "";
+
+	// For form and meter, both are nested within other component package.json files
+  let nestedComponent = packageJson?.nestedComponentName ?? undefined;
+	
+	if(linkType === "package") {
+		// NPM package name and link 
+		packageName = packageJson?.name ?? undefined;
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://npmjs.com/${packageName}` : false;
+	}
+	
+	else if (linkType === "repository") {
+		// repo name and link 
+		packageName = packageJson?.name ? packageJson?.name.split('/').pop() : undefined;
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://github.com/adobe/spectrum-css/tree/main/components/${packageName}` : false;
+	}
+
+	else if (linkType === "guidelines") {
+		// guidelines site name and link 
+		packageName = packageJson?.componentGuidelinesName ?? undefined;
+
+		 // TODO: This may not be a sustainable approach to targeting specific nested components. For example, text area is sort of nested under text field, but we don't surface text area as a separate component, like meter or form. We should probably refactor this to either support nested components more dynamically or potentially un-nest components.
+		if (nestedComponent === "form") {
+			packageName = undefined;
+		}
+
+		if (nestedComponent === "meter") {
+			packageName = nestedComponent;
+		}
+
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://spectrum.adobe.com/page/${packageName}` : false;
+
+		// internal contributions/beta guidelines name and link
+		packageAltName = (packageName) ? undefined : packageJson?.componentBetaName;
+
+		if (!packageLink) {
+			packageAltLink = (packageAltName && typeof packageAltName !== "undefined") ? `https://spectrum-contributions.corp.adobe.com/page/${packageAltName}` : false;
+			packageLink = packageAltLink;
+		}
+	}
```

Comment:
It definitely feels like there's probably room to refactor this. I'm not a React expert, but some of it feels repetitive. If anybody has ideas on how to more efficiently get the package name and create the right links, I'm all ears!